kafka_metric/getmessages_kafka_metric.py

```diff
@@ -1035,18 +1090,22 @@ def main():
     # all processes
     processes = []
 
-    # consumer process
-    for x in range(multiprocessing.cpu_count()):
-        d = Process(target=process_get_data,
-                    args=(log_queue, cli_config_vars, if_config_vars, agent_config_vars, messages))
-        d.daemon = True
-        d.start()
-        processes.append(d)
-
+    for kafka_connect in agent_config_vars['kafka_connect_file']:
+        kafka_info = get_multiple_kafka_info(logger, kafka_connect, if_config_vars)
+        agent_config_vars.update(kafka_kwargs=kafka_info['kafka_kwargs'], topics=kafka_info['topics'])
+        # consumer process
+        for x in range(multiprocessing.cpu_count()):
```

Comment:
This needs to be configurable rather than being dependent on the number of CPUs. The max kafka consumers you can use on a system is equal to the number of kafka partitions enabled, so it doesn't help to have 8 consumers if you only have 4 topics.

Reply:
The customer implementation has the same number of partitions for all clusters; however, it might vary, so I think we should move the consumer specification into the kafka_config.ini rather than the default configuration.